src/coecms/um.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+# Copyright 2018 ARC Centre of Excellence for Climate Extremes
+# author: Scott Wales <scott.wales@unimelb.edu.au>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import print_function
+
+from coecms.dimension import identify_lat_lon, identify_time
+
+import mule
+import numpy
+import xarray
+import dask.array
+
+
+def create_surface_ancillary(input_ds, stash_map):
+    """Create a surface-level UM ancillary file
+
+    Args:
+        input_ds: Source dataset/dataarray
+        output_filename: UM ancillary file to create
+        stash_map: Mapping of variable name from `input_ds` to STASH code
+
+    Returns:
+        :obj:`mule.AncilFile` containing ancillary file data, write out with
+        ``.to_file()``
+
+    Example:
+        ::
+
+            input_ds = xarray.open_mfdataset(files, engine='pynio')
```

Comment:
Is there a reason to use `pynio`? Just curious

Reply:
I was looking at using this to convert GRIB data, which needs pynio. Not needed for netcdf